pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +755,97 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	copyNameNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "name"},
+	)
+	copyDiffNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "diff"},
+	)
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileName,
+		OnPress: func() error {
+			name := self.context().GetSelectedItemId()
+			if name == "" {
+				return self.c.ErrorMsg(copyNameNoItemSelectionErr)
+			}
+
+			return self.copyToClipboard(name)
+		},
+		Key: 'n',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label: self.c.Tr.CopySelectedDiff,
+		OnPress: func() error {
+			path := self.context().GetSelectedPath()
+			if path == "" {
+				return self.c.ErrorMsg(copyDiffNoItemSelectionErr)
+			}
+			diff, err := self.c.Git().Diff.GetPathDiff(path, self.hasPathStagedChanges())
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 's',
+	}
+
+	if self.context().GetSelected() == nil {
+		copyNameItem.DisabledReason = copyNameNoItemSelectionErr
+		copyFileDiffItem.DisabledReason = copyDiffNoItemSelectionErr
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.CopyToClipboardMenu,
+		Items: []*types.MenuItem{
```

Comment:
I remember a discussion we had on a previous PR, where we decided it was better to avoid hiding elements in favor of disabling/explicitly throwing errors to make the UI cleaner. I may be wrong, but I think it's better to give some feedback instead of disabling the shortcut/hiding the feature. I'd prefer to keep this as is if you're happy with it.

Reply:
I agree here. We should have all commands visible, and disable the ones that are not applicable. In the case of a menu like this, it can result in _all_ menu items being disabled, which I think is fine. It helps new users learn what commands exist.